flash : stm32l4x don't write 0xff

jamike opened the pull request:
for mcuboot on l4, avoiding writting 0xff solves trailer write access during swap

galak reviewed (changes requested):
Any reason not to include this in PR #388?

Also similar comment about subject of commit message provide additional details about the issue & fix.

lpereira commented:
You might want to rebase this, so it only includes the last patch. You can force-push to your repository, this PR will be updated.

Commit:
pinmux: pinmux_dev_k64 driver and related references are removed.

Functionality of a pinmux driver is now a part of a regular
driver.

Signed-off-by: Shiksha Patel <shiksha.patel@nxp.com>

jamike commented:
This commit is required is required only for MCU boot loader. That 's why I pushed it separetly.

erwango commented on drivers/flash/flash_stm32l4x.c:
```diff
@@ -142,9 +142,13 @@ int flash_stm32_write_range(unsigned int offset, const void *data,
+		if (((const u64_t *) data)[i>>3] != -1) {
+			rc = write_dword(offset,
+					((const u64_t *)data)[i>>3]
+					, p);
```
can you move coma at end of second line?

erwango reviewed (changes requested):
minor comment, otherwise LGTM

galak committed:
arm: Add build time consistency check for irq priority defines

We need to make sure that __NVIC_PRIO_BITS & CONFIG_NUM_IRQ_PRIO_BITS
are set to the same value.  Add a simple build time check to ensure
this is the case.  This is to catch future cases of issues like
ZEP-2243.  This is a stop gap til we resolve ZEP-2262, which covers use
of both __NVIC_PRIO_BITS & CONFIG_NUM_IRQ_PRIO_BITS.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

jamike committed:
flash: stm32: l4x don't write 0xff

As l4 implements ECC on 8 flash bytes (64 bits),
writing the 8 bytes to 0xff, triggers the write
of ECC bits,and prevents from beeing able to
write these 8 bytes later to another value.
Current MCU boot loader uses this facility to
write swap status in trailer field while moving
last sector to scratch sector and then
to slot sector.

Signed-off-by: Michel Jaouen <michel.jaouen@st.com>

galak commented:
now that the other flash changes are merged, can you rebase/update this.

jamike commented:
Following some changes in mcu bootloader . this patch seems not needed . I continue testing without. 
And if confirmed , i will close it.

galak commented:
any updates if still needed?

jamike commented:
no

jamike commented:
After the changes in mcuboot , the PR is not needed any more.
